llvm/lib/Transforms/Utils/CloneFunction.cpp

```diff
@@ -700,6 +697,16 @@ void llvm::CloneAndPruneIntoFromInst(Function *NewFunc, const Function *OldFunc,
     }
   }
 
+  // Drop all incompatible return attributes that cannot be applied to NewFunc
+  // during cloning, so as to allow instruction simplification later to reason
+  // on the old state of the function. The original attributes are restored
+  // before returning.
+  AttributeMask IncompatibleAttrs =
+      AttributeFuncs::typeIncompatible(OldFunc->getReturnType());
+  AttrBuilder RetAttrs(NewFunc->getContext(),
+                       NewFunc->getAttributes().getRetAttrs());
+  NewFunc->removeRetAttrs(IncompatibleAttrs);
```

Comment:
Maybe move this directly before the code doing the actual simplification?

Reply:
Perhaps we should drop args attribute as well, for coherence?